core/types/transaction.go

```diff
@@ -125,18 +142,127 @@ func isProtectedV(V *big.Int) bool {
 
 // EncodeRLP implements rlp.Encoder
 func (tx *Transaction) EncodeRLP(w io.Writer) error {
+	if tx.data.FeeCap == nil || tx.data.GasPremium == nil {
+		return rlp.Encode(w, []interface{}{
+			tx.data.AccountNonce,
+			tx.data.Price,
+			tx.data.GasLimit,
+			tx.data.Recipient,
+			tx.data.Amount,
+			tx.data.Payload,
+			tx.data.V,
+			tx.data.R,
+			tx.data.S,
+		})
+	}
 	return rlp.Encode(w, &tx.data)
 }
 
 // DecodeRLP implements rlp.Decoder
-func (tx *Transaction) DecodeRLP(s *rlp.Stream) error {
-	_, size, _ := s.Kind()
-	err := s.Decode(&tx.data)
-	if err == nil {
+func (tx *Transaction) DecodeRLP(stream *rlp.Stream) error {
```

Comment:
What we've historically done in these kinds of situations, is to first attempt to decode into one, and then the other. E.g. 
```
var newTx Eip1559Transaction 
var oldTx Transaction 
if err := s.Decode( &newTx); err != nil{
   err = s.Decode(&oldTx)
}
```

Reply:
I recall trying this approach originally and running into issues but it's been so long that I don't recall the exact nature of them. A quick conversion to this is yielding errors in unit tests but I need to look into it more!